tests/virt_autotest/login_console_as_kvm_or_xen_role.pm

```diff
@@ -0,0 +1,66 @@
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: For openSUSE virtualization test only. login in console, install kvm/xen patterns if needed.
+# Maintainer: Julie CAO <jcao@suse.com>
+
+use strict;
+use warnings;
+use base 'virt_autotest_base';
+use testapi;
+use ipmi_backend_utils;
+use Utils::Backends qw(use_ssh_serial_console is_remote_backend);
+use utils qw(zypper_call systemctl);
+use virt_autotest::utils qw(is_kvm_host is_xen_host);
+
+sub run {
+
+    #local openQA can skip TW host install
+    if (get_var('IPMI_DO_NOT_RESTART_HOST')) {
+        select_console 'sol', await_console => 0;
+        send_key 'ret';
+    }
+
+    use_ssh_serial_console !check_screen('text-logged-in-root');
+    return if get_var('IPMI_DO_NOT_RESTART_HOST');
+
+    zypper_call("--gpg-auto-import-keys ref",         180);
+    if (is_kvm_host) {
+        zypper_call("in -t pattern kvm_server kvm_tools", 900);     #for TW, tried 600 sec is not enough
+        save_screenshot;
+        set_serial_console_on_vh('', '', 'kvm');
+    }
+    if (is_xen_host) {
+        zypper_call("in -t pattern xen_server xen_tools", 1800);
+        save_screenshot;
+        set_serial_console_on_vh('', '', 'xen');
+    }
+
+    virt_autotest::utils::install_default_packages();
+
+    #libvirtd failed to be started in Tumbleweed
+    #not sure if it is a bug yet, just work it around here to make tests run
+    systemctl 'status libvirtd', ignore_failure => 1;
+    save_screenshot;
+    script_run "echo DEBUG:; which dnsmasq; which iptables; which ip6tables; which ebtables";
+    save_screenshot;
+    zypper_call("in dnsmasq iptables bridge-utils", 1800);
+    script_run "ln /usr/sbin/dnsmasq /sbin/dnsmasq";
+    script_run "ln /usr/sbin/iptables /sbin/iptables";
+    script_run "ln /usr/sbin/ip6tables /sbin/ip6tables";
+    script_run "ln /usr/sbin/ebtables /sbin/ebtables";
```

Comment:
Why do you use hard-links? Please use symlinks by default. Also use `assert_script_run` unless you want to explicitly check the return value:

```suggestion
    assert_script_run("ln -s /usr/sbin/$_ /sbin/$_") for qw(dnsmasq iptables ip6tables ebtables);
```

but why do it at all? Commands should be called from $PATH by default

Reply:
Did not notice hard-links was used ~ The value of $PATH looks ok but it does not work here(and other place sometimes). libvirtd failed to be started for lack of /sbin/xxx. I have not located the cause, not sure it is libvirtd problem or $PATH problem. I am going to open a bug later. It is a workaround before the issue gets fixed.

Reply:
If it is workaround, please add a comment about it.

Reply:
comments have been added.
   #libvirtd failed to be started in Tumbleweed
    #not sure if it is a bug yet, just work it around here to make tests run